pkg/services/tree/sync.go

```diff
@@ -85,15 +86,36 @@ func (s *Service) synchronizeAllTrees(ctx context.Context, cid cid.ID) error {
 		return fmt.Errorf("could not fetch tree ID list: %w", outErr)
 	}
 
+	s.cnrMapMtx.Lock()
+	oldStatus := s.cnrMap[cid]
+	s.cnrMapMtx.Unlock()
+
+	syncStatus := map[string]uint64{}
+	for i := range treesToSync {
+		syncStatus[treesToSync[i]] = 0
+	}
+	for tid := range oldStatus {
+		if _, ok := syncStatus[tid]; ok {
+			syncStatus[tid] = oldStatus[tid]
+		}
+	}
+
 	for _, tid := range treesToSync {
-		err = s.synchronizeTree(ctx, d, tid, nodes)
+		h, err := s.synchronizeTree(ctx, d, syncStatus[tid], tid, nodes)
 		if err != nil {
 			s.log.Error("could not sync tree",
 				zap.Stringer("cid", cid),
 				zap.String("treeID", tid))
 		}
+		if syncStatus[tid] < h {
```

Comment:
can we remember `syncStatus[tid]` in a local var and not read `map` twice?

Reply:
It is more complex IMO and it won't be any faster because network requests take a lot more time to execute.